pkg/verifier/notation/notation.go

```diff
@@ -63,37 +64,50 @@ type NotationPluginVerifierConfig struct { //nolint:revive // ignore linter to h
 }
 
 type notationPluginVerifier struct {
+	name             string
+	typeName         string
 	artifactTypes    []string
 	notationVerifier *notation.Verifier
 }
 
 type notationPluginVerifierFactory struct{}
 
 func init() {
-	factory.Register(verifierName, &notationPluginVerifierFactory{})
+	factory.Register(verifierTypeName, &notationPluginVerifierFactory{})
 }
 
 func (f *notationPluginVerifierFactory) Create(_ string, verifierConfig config.VerifierConfig, pluginDirectory string, namespace string) (verifier.ReferenceVerifier, error) {
 	logger.GetLogger(context.Background(), logOpt).Debugf("creating notation with config %v, namespace '%v'", verifierConfig, namespace)
+	verifierName := fmt.Sprintf("%s", verifierConfig[types.Name])
```

Comment:
I feel we might need some validation on `verifierName` and `verifierType`. e.g.
If users only provide `verifierName`, then it must be `notation`. And if users provide both `name` and `type`, then the `type` must be `notation`.

Reply:
Discussion in PR meeting, this validation is done somewhere else

Reply:
factory.go